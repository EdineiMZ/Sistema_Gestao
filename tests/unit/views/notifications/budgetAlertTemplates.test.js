const path = require('path');
const ejs = require('ejs');

describe('views/notifications budget alert templates', () => {
    const templatesDir = path.join(__dirname, '../../../../src/views/notifications');
    const now = new Date('2024-05-10T12:00:00Z');

    const budgetSample = {
        budgetId: 21,
        categoryName: 'Marketing Digital',
        monthlyLimit: 15000,
        consumption: 13850.75,
        month: '2024-05',
        thresholds: [10000, 13000],
        categoryColor: '#f97316'
    };

    const baseOptions = {
        user: { id: 77, name: 'Ana Paula Souza' },
        baseUrl: 'https://app.exemplo.com',
        routePath: '/finance/orcamentos',
        organizationName: 'Sistema de Gestão Inteligente',
        appName: 'Sistema de Gestão Inteligente',
        now
    };

    let emailUtils;

    beforeAll(() => {
        process.env.BUDGET_LINK_SECRET = 'unit-test-secret';
        jest.resetModules();
        // eslint-disable-next-line global-require
        emailUtils = require('../../../../src/utils/email');
    });

    afterAll(() => {
        delete process.env.BUDGET_LINK_SECRET;
        jest.resetModules();
    });

    it('gera snapshot consistente do template de e-mail', async () => {
        const context = emailUtils.buildBudgetAlertContext(budgetSample, baseOptions);
        const html = await ejs.renderFile(
            path.join(templatesDir, 'budgetAlertEmail.ejs'),
            context,
            { async: true }
        );

        expect(html.replace(/\s+/g, ' ').trim()).toMatchSnapshot();
    });

    it('gera snapshot consistente do template in-app', async () => {
        const context = emailUtils.buildBudgetAlertContext(budgetSample, baseOptions);
        const html = await ejs.renderFile(
            path.join(templatesDir, 'budgetAlertInApp.ejs'),
            context,
            { async: true }
        );

        expect(html.replace(/\s+/g, ' ').trim()).toMatchSnapshot();
    });
});
