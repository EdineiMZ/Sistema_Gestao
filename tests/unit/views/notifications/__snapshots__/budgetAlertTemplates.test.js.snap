// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/notifications budget alert templates gera snapshot consistente do template de e-mail 1`] = `"<!DOCTYPE html> <html lang="pt-BR"> <head> <meta charset="UTF-8" /> <meta http-equiv="X-UA-Compatible" content="IE=edge" /> <meta name="viewport" content="width=device-width, initial-scale=1.0" /> <title>Orçamento de Marketing Digital</title> <style> body { margin: 0; padding: 0; background-color: #f5f7fb; font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; color: #0f172a; } .email-wrapper { width: 100%; padding: 32px 0; } .email-card { max-width: 680px; margin: 0 auto; background: #ffffff; border-radius: 24px; overflow: hidden; box-shadow: 0 28px 60px rgba(15, 23, 42, 0.18); } .email-hero { background: #f97316; color: #ffffff; padding: 36px 44px 32px; } .email-hero span.label { display: inline-flex; align-items: center; gap: 8px; font-size: 12px; letter-spacing: 0.12em; text-transform: uppercase; font-weight: 600; opacity: 0.9; } .email-hero h1 { margin: 16px 0 8px; font-size: 30px; font-weight: 600; line-height: 1.2; } .email-hero p { margin: 0; font-size: 16px; line-height: 1.6; max-width: 520px; } .email-body { padding: 36px 44px; line-height: 1.6; } .email-body p { margin: 0 0 18px; color: #475569; font-size: 15px; } .email-body ul { margin: 24px 0; padding-left: 20px; color: #475569; font-size: 15px; } .email-body li { margin-bottom: 10px; } .cta-button { display: inline-block; margin-top: 8px; padding: 14px 28px; border-radius: 999px; background: #f97316; color: #ffffff !important; text-decoration: none; font-weight: 600; box-shadow: 0 16px 32px rgba(15, 23, 42, 0.18); transition: transform 0.2s ease, box-shadow 0.2s ease; } .cta-button:hover { transform: translateY(-2px); box-shadow: 0 20px 36px rgba(15, 23, 42, 0.22); } .email-footer { background: #f1f5f9; padding: 26px 44px; font-size: 13px; color: #64748b; line-height: 1.6; } @media (max-width: 640px) { .email-card { margin: 0 16px; } .email-hero, .email-body, .email-footer { padding: 28px; } .email-hero h1 { font-size: 26px; } } </style> </head> <body> <span style="display:none;color:transparent;height:0;max-height:0;opacity:0;overflow:hidden;"> 92,3% do limite utilizado Categoria próxima do limite configurado. Reavalie os gastos recentes. </span> <div class="email-wrapper"> <div class="email-card"> <div class="email-hero"> <span class="label"><i>●</i> Alerta financeiro</span> <h1>Orçamento de Marketing Digital</h1> <p>Categoria próxima do limite configurado. Reavalie os gastos recentes.</p> </div> <div class="email-body"> <p>Olá <strong>Ana</strong>,</p> <p> Detectamos uma atualização importante no orçamento da categoria <strong>Marketing Digital</strong>. </p> <div class="budget-alert-summary-email" style="border:1px solid #e2e8f0;border-radius:18px;padding:24px;background:#ffffff;box-shadow:0 18px 36px rgba(15,23,42,0.08);" data-budget-alert-summary > <div class="budget-alert-summary-header"> <div class=""> <span class="" style="display:inline-flex;align-items:center;gap:6px;font-size:12px;font-weight:600;text-transform:uppercase;letter-spacing:0.08em;padding:6px 12px;border-radius:999px;background:#fef3c7;color:#b45309;" > Atenção ao consumo </span> <h3 class="" style="margin:16px 0 4px;font-size:20px;font-weight:600;color:#0f172a;" > Orçamento de Marketing Digital </h3> <p class="" style="margin:0 0 12px;font-size:14px;color:#475569;" > Referência: <strong>maio de 2024</strong> </p> <p class="" style="margin:0;color:#475569;font-size:14px;" > 92,3% do limite utilizado </p> </div> <div class="" style="margin-top:20px;width:100%;" > <div class="" style="width:100%;height:12px;border-radius:999px;background:#e2e8f0;overflow:hidden;margin-bottom:12px;" > <div class="" style="height:100%;width:92.3%;background:#f97316;" role="progressbar" aria-valuenow="92.3" aria-valuemin="0" aria-valuemax="150" ></div> </div> <span class="" style="display:block;font-weight:600;color:#0f172a;font-size:14px;" > R$ 13.850,75 consumidos </span> </div> </div> <div class="" style="margin-top:20px;display:flex;gap:16px;flex-wrap:wrap;" > <div class="" style="flex:1 1 160px;"> <div class="" style="border:1px solid #e2e8f0;border-radius:14px;padding:16px;background:#f8fafc;height:100%;" > <span class="" style="display:block;font-size:12px;font-weight:600;text-transform:uppercase;letter-spacing:0.06em;color:#64748b;margin-bottom:8px;" > Limite mensal </span> <span class="" style="display:block;font-size:18px;font-weight:600;color:#0f172a;" > R$ 15.000,00 </span> </div> </div> <div class="" style="flex:1 1 160px;"> <div class="" style="border:1px solid #e2e8f0;border-radius:14px;padding:16px;background:#f8fafc;height:100%;" > <span class="" style="display:block;font-size:12px;font-weight:600;text-transform:uppercase;letter-spacing:0.06em;color:#64748b;margin-bottom:8px;" > Consumido no período </span> <span class="" style="display:block;font-size:18px;font-weight:600;color:#0f172a;" > R$ 13.850,75 </span> </div> </div> <div class="" style="flex:1 1 160px;"> <div class="" style="border:1px solid #e2e8f0;border-radius:14px;padding:16px;background:#f8fafc;height:100%;" > <span class="" style="display:block;font-size:12px;font-weight:600;text-transform:uppercase;letter-spacing:0.06em;color:#64748b;margin-bottom:8px;" > Disponível </span> <span class="" style="display:block;font-size:18px;font-weight:600;color:#047857;" > R$ 1.149,25 </span> <span class="" style="display:block;margin-top:8px;font-size:13px;color:#64748b;line-height:1.5;" > Ainda há saldo para novas movimentações neste mês. </span> </div> </div> </div> <div class="" style="margin-top:24px;padding:16px 20px;border-radius:14px;background:#fffbeb;border:1px solid #facc15;color:#92400e;font-size:14px;line-height:1.5;" > Limite de atenção configurado em <strong>R$ 13.000,00</strong> (86,7% do limite). </div> </div> <ul> <li>92,3% do limite utilizado</li> <li>O alerta configurado em R$ 13.000,00 já foi atingido.</li> <li>Ainda restam R$ 1.149,25 disponíveis neste orçamento.</li> <li>Categoria próxima do limite configurado. Reavalie os gastos recentes.</li> </ul> <a class="cta-button" href="https://app.exemplo.com/finance/orcamentos?budgetId=21&amp;budgetToken=eyJiaWQiOjIxLCJ1aWQiOjc3LCJpYXQiOjE3MTUzNDI0MDAsImV4cCI6MTcxNTQyODgwMH0.BUna5Td1QVgWOZDr2_UFD2-FKzu44z9mtI6k-qR1tVs">Acessar orçamento</a> <p style="margin-top: 28px; font-size: 13px;"> Link seguro válido até <strong>11/05/2024, 09:00</strong>. Não compartilhe este endereço: ele é exclusivo da sua conta e leva diretamente ao módulo de orçamentos. </p> </div> <div class="email-footer"> <p style="margin: 0 0 8px;"><strong>Sistema de Gestão Inteligente</strong></p> <p style="margin: 0;"> Você recebeu este alerta automático do módulo financeiro. Caso não reconheça a operação, revise as permissões de acesso e entre em contato com o time responsável. </p> </div> </div> </div> </body> </html>"`;

exports[`views/notifications budget alert templates gera snapshot consistente do template in-app 1`] = `"<div class="card card-soft shadow-sm h-100" data-budget-alert> <div class="card-body"> <div class="d-flex flex-column flex-lg-row justify-content-between align-items-lg-center gap-3 mb-3"> <div> <span class="app-chip mb-2"><i class="bi bi-graph-up-arrow me-2"></i>Orçamentos</span> <h4 class="fw-semibold mb-1">Orçamento de Marketing Digital</h4> <p class="text-muted mb-0">Categoria próxima do limite configurado. Reavalie os gastos recentes.</p> </div> <div class="text-muted small text-lg-end"> Atualizado em <strong>10/05/2024, 09:00</strong> </div> </div> <div class="budget-alert-summary card-soft border-0" style="" data-budget-alert-summary > <div class="d-flex flex-column flex-lg-row align-items-lg-center gap-4 mb-3"> <div class="flex-grow-1"> <span class="badge bg-warning-subtle text-warning" style="" > Atenção ao consumo </span> <h3 class="h5 fw-semibold mt-3 mb-1 text-dark" style="" > Orçamento de Marketing Digital </h3> <p class="text-muted mb-2" style="" > Referência: <strong>maio de 2024</strong> </p> <p class="text-muted mb-0" style="" > 92,3% do limite utilizado </p> </div> <div class="text-lg-end w-100" style="" > <div class="progress bg-light rounded-pill mb-2" style="" > <div class="progress-bar" style="width:92.3%;background:#f97316;" role="progressbar" aria-valuenow="92.3" aria-valuemin="0" aria-valuemax="150" ></div> </div> <span class="fw-semibold text-dark small" style="" > R$ 13.850,75 consumidos </span> </div> </div> <div class="row g-3 mt-3" style="" > <div class="col-12 col-sm-4" style=""> <div class="border rounded-4 px-3 py-3 bg-light-subtle h-100" style="" > <span class="text-muted text-uppercase small fw-semibold d-block mb-1" style="" > Limite mensal </span> <span class="fw-semibold text-dark fs-5" style="" > R$ 15.000,00 </span> </div> </div> <div class="col-12 col-sm-4" style=""> <div class="border rounded-4 px-3 py-3 bg-light-subtle h-100" style="" > <span class="text-muted text-uppercase small fw-semibold d-block mb-1" style="" > Consumido no período </span> <span class="fw-semibold text-dark fs-5" style="" > R$ 13.850,75 </span> </div> </div> <div class="col-12 col-sm-4" style=""> <div class="border rounded-4 px-3 py-3 bg-light-subtle h-100" style="" > <span class="text-muted text-uppercase small fw-semibold d-block mb-1" style="" > Disponível </span> <span class="fw-semibold text-success fs-5" style="" > R$ 1.149,25 </span> <span class="text-muted small d-block mt-1" style="" > Ainda há saldo para novas movimentações neste mês. </span> </div> </div> </div> <div class="alert alert-warning border-0 shadow-sm mt-4 mb-0" style="" > <i class="bi bi-exclamation-triangle-fill me-2" aria-hidden="true"></i> Limite de atenção configurado em <strong>R$ 13.000,00</strong> (86,7% do limite). </div> </div> <ul class="text-muted small ps-3 mt-4 mb-4"> <li>92,3% do limite utilizado</li> <li>O alerta configurado em R$ 13.000,00 já foi atingido.</li> <li>Ainda restam R$ 1.149,25 disponíveis neste orçamento.</li> <li>Categoria próxima do limite configurado. Reavalie os gastos recentes.</li> </ul> <div class="d-flex flex-column flex-sm-row align-items-sm-center justify-content-between gap-3 mt-4"> <div class="text-muted small"> Link seguro válido até <strong>11/05/2024, 09:00</strong> </div> <a class="btn btn-gradient" href="/finance/orcamentos?budgetId=21&amp;budgetToken=eyJiaWQiOjIxLCJ1aWQiOjc3LCJpYXQiOjE3MTUzNDI0MDAsImV4cCI6MTcxNTQyODgwMH0.BUna5Td1QVgWOZDr2_UFD2-FKzu44z9mtI6k-qR1tVs"> <i class="bi bi-arrow-right-short me-1" aria-hidden="true"></i> Ir para orçamentos </a> </div> </div> </div>"`;
